iterators/local.js

```diff
@@ -0,0 +1,47 @@
+module.exports = class LocalBlocksIterator {
+  constructor (batch, opts = {}) {
+    this.batch = batch
+    this.options = opts
+    this.gte = 0
+    this.lt = 0
+    this.limit = typeof opts.limit === 'number' ? opts.limit : -1
+  }
+
+  async open () {
+    await this.batch.getRoot(false) // does the update dance
+    this.gte = gte(this.options, this.batch.version)
+    this.lt = lt(this.options, this.batch.version)
+  }
+
+  async next () {
+    if (this.limit === 0) return null
+    if (this.limit > 0) this.limit--
+
+    while (this.gte < this.lt) {
+      try {
+        return await this.batch.getBlock(this.gte++)
+      } catch {
+        continue
+      }
+    }
+
+    return null
+  }
+
+  close () {
+    return this.batch._closeSnapshot()
```

Comment:
(detail) Why not call `this.batch.close()` here? Then it's the public API (it's the same, since the first line in `close` is `if (this.isSnapshot) return this._closeSnapshot`